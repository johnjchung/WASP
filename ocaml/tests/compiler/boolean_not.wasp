Endpoint boolOps(GET:) {
	bool a = not true;
	bool b = not false;
}
