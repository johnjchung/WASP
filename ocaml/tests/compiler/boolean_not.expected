package main

import (
	"github.com/gin-gonic/gin"
	"math"
	"os"
)

func int_conv(anything interface{}) int {
	switch v := anything.(type) {
		case int:
			return v
		case float64, float32:
			var float_var = v.(float64)
			return int(math.Trunc(float_var))
		default:
			return 0
	}
	return 0
}

func float_conv(anything interface{}) float64 {
 	 switch v := anything.(type) {
		case int:
            return float64(v)
		case float64:
			return v
		default:
			return 0
	}
	return 0.0
}

type boolOps_struct struct {
	A bool `db:"a" json:"a"`
	B bool `db:"b" json:"b"`
}

func main() {
	router := gin.Default()
	router.GET("/boolOps", Get_boolOps)
	router.Run(":"+os.Getenv("PORT"))
}

func boolOps_struct_wrapper() boolOps_struct {
	return boolOps_struct{A: bool(!true), B: bool(!false)}
}

func Get_boolOps(c *gin.Context) {
	boolOps_struct := boolOps_struct_wrapper()
	c.JSON(200, boolOps_struct)
}
